core/http/protocol/src/main/java/org/eclipse/rdf4j/http/protocol/Protocol.java

```diff
@@ -181,9 +181,9 @@
 	public static final String ACCEPT_PARAM_NAME = "Accept";
 
 	/**
-	 * Parameter name for the isolation level used in transactions.
+	 * Prefix for transaction settings in the query param
 	 */
-	public static final String ISOLATION_LEVEL_PARAM_NAME = "isolation-level";
+	public static final String TRANSACTION_SETTINGS_PREFIX = "transaction-setting__";
```

Comment:
Removing a parameter from the public REST API and replacing it with something else is backward incompatible. I think the simplest solution is that at the level of the REST API protocol we keep IsolationLevel and other Transaction Settings distinct.

Reply:
Actually what I ended up doing for this is that the client-side sends the isolation level twice (once using the legacy param, once with the name prefixed param name), and the server-side accepts both as well. This way, the change is backward compatible for both client and server: an older client communicating with a newer server, or vice versa.

Reply:
I was hoping our rest api wasn’t part of the semantic versioning contract. Ie. that it was considered an internal api between the server and client.

Reply:
It most definitely is I'm afraid. We have a full canonical spec published for it.